pkg/apis/apps/types.go

```diff
@@ -44,6 +44,21 @@ type StatefulSet struct {
 	Status StatefulSetStatus
 }
 
+// PodCreationPolicyType defines the policy for creating pods under a stateful set.
+type PodCreationPolicyType string
+
+const (
+	// OrderedReadyPodCreation will create pods in strictly increasing order on
+	// scale up and strictly decreasing order on scale down, progressing only when
+	// the previous pod is ready or terminated. At most one pod will be changed
+	// at any time.
+	OrderedReadyPodCreation PodCreationPolicyType = "OrderedReady"
+	// ParallelPodCreation will create and delete pods as soon as the stateful set
```

Comment:
Update or pod is deleted by a pod drain.  A user probably doesn't always think of that as "creation" because they didn't explicitly ask for it.  "Scaling" means something very precise in our API - changing the replica count.  The responsibilities of the controller outside of that are not really "scaling".